util/log/file.go

```diff
@@ -221,3 +296,92 @@ func GetLogReader(filename string, allowAbsolute bool) (io.ReadCloser, error) {
 	}
 	return nil, err
 }
+
+// int64sortable is required so we can sort on an int64 slice.
+type int64sortable []int64
+
+func (a int64sortable) Len() int           { return len(a) }
+func (a int64sortable) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a int64sortable) Less(i, j int) bool { return a[i] < a[j] }
+
+// FetchEntiresFromFiles fetches all available logs on disk that are of the
+// level of severity (or worse) and are between the start and end times. It will
+// stop reading in new files if the EntiresCutoff is exceeded, no more files
+// will be retrieved. The logs entries returned will be in decreasing order,
+// with the closest log to the start time being the first entry.
+func FetchEntiresFromFiles(level Level, startTimeNano, endTimeNano int64) ([]proto.LogEntry, error) {
+	logFiles, err := ListLogFiles()
+	if err != nil {
+		return nil, err
+	}
+
+	logFileMap := make(map[int64]FileInfo)
+	var nanos int64sortable
+	var diffToStart int64 = math.MaxInt64
+	// closestToStartNano holds the last log file that would include any entries
+	// from the start time onward in it.
+	var closestToStartNano int64
+	for _, logFile := range logFiles {
+		if logFile.Details.Level == level {
```

Comment:
This is just a preference, but in cases like to use `continue` statements to avoid excessive indentation.

```
if logFile.Details.Level != level {
   continue;
}
```